### PR TITLE
[Draft]Prepare idl change for workflow isolation

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**

**Why?**

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/dataManagerInterfaces.go [hunk 1]
```diff
@@ -343,6 +343,7 @@ type (
 		AutoResetPoints                    *types.ResetPoints
 		Memo                               map[string][]byte
 		SearchAttributes                   map[string][]byte
+		PartitionConfig                    map[string]string
```
Comment (davidporter-id-au) [inline common/persistence/dataManagerInterfaces.go:346]: I was considering type-aliasing the PartitionConfig just to make any refactoring easier, but this is also fine imho

File: common/persistence/serialization/thrift_mapper_test.go [hunk 2]
```diff
@@ -214,6 +214,8 @@ func TestWorkflowExecutionInfo(t *testing.T) {
 		Memo:                               map[string][]byte{"key_1": []byte("Memo")},
 		VersionHistories:                   []byte("VersionHistories"),
 		VersionHistoriesEncoding:           "VersionHistoriesEncoding",
+		FirstExecutionRunID:                UUID(uuid.New()),
+		PartitionConfig:                    map[string]string{"zone": "dca1"},
```
Comment (davidporter-id-au) [inline common/persistence/serialization/thrift_mapper_test.go:218]: do you mind adding a mapper test where this is nil too?

**Review by davidporter-id-au — APPROVED** [review]
LGTM but minor request for unit-testing in mapping for nils

File: common/persistence/serialization/thrift_mapper_test.go (hunk 2 above)
Comment (Shaddoll) [reply common/persistence/serialization/thrift_mapper_test.go:218]: It's already coverred.
